File: searchBar/src/main/java/com/example/utils/DimensionUtils.java
package com.example.utils;

import android.content.Context;
import android.util.DisplayMetrics;

import org.jetbrains.annotations.NotNull;

public final class DimensionUtils {

    public static int pxToDp(int px, @NotNull Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        return Math.round(px / (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT));
    }

    public static int dpToPx(int dp, @NotNull Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        return Math.round(dp * (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT));
    }
}

File: searchBar/src/main/java/com/example/utils/DimensionUtils.kt
package com.example.utils

import android.content.Context
import android.util.DisplayMetrics
import kotlin.math.roundToInt

object DimensionUtils {
    fun pxToDp(px: Int, context: Context): Int {
        val displayMetrics = context.resources.displayMetrics
        return (px / (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT)).roundToInt()
    }

    fun dpToPx(dp: Int, context: Context): Int {
        val displayMetrics = context.resources.displayMetrics
        return (dp * (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT)).roundToInt()
    }
}